2.0/m_conn_matchident.cpp

```diff
@@ -0,0 +1,60 @@
+/*
+ * InspIRCd -- Internet Relay Chat Daemon
+ *
+ *   Copyright (C) 2017 genius3000 <genius3000@g3k.solutions>
+ *
+ * This file is part of InspIRCd.  InspIRCd is free software: you can
+ * redistribute it and/or modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation, version 2.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/* $ModAuthor: genius3000 */
+/* $ModAuthorMail: genius3000@g3k.solutions */
+/* $ModDesc: Allows a connect class to match by ident. */
+/* $ModDepends: core 2.0 */
+/* $ModConfig: Within connect block: matchident="myIdent" */
+
+
+#include "inspircd.h"
+
+class ModuleConnMatchIdent : public Module
+{
+ public:
+	void init()
+	{
+		Implementation eventlist[] = { I_OnSetConnectClass };
+		ServerInstance->Modules->Attach(eventlist, this, sizeof(eventlist)/sizeof(Implementation));
```

Comment:
If you're only hooking to one event you can just use the non-array form of Attach.

Reply:
Good point, fixed.